core/services/ardupilot_manager/firmware/FirmwareInstall.py

```diff
@@ -0,0 +1,104 @@
+import json
+import os
+import pathlib
+import shutil
+import stat
+
+from exceptions import (
+    FirmwareInstallFail,
+    InvalidFirmwareFile,
+    UndefinedPlatform,
+    UnsupportedPlatform,
+)
+from firmware.FirmwareDownload import FirmwareDownloader, FirmwareFormat, Platform
+from firmware.FirmwareUpload import FirmwareUploader
+
+
+def get_board_id(platform: Platform) -> int:
+    ardupilot_board_ids = {
+        Platform.Pixhawk1: 9,
+    }
+    return ardupilot_board_ids.get(platform, -1)
+
+
+class FirmwareInstaller:
+    """Abstracts the install procedures for different supported boards.
+
+    For proper usage one needs to set the platform before using other methods.
+
+    Args:
+        firmware_folder (pathlib.Path): Path for firmware folder.
+    """
+
+    def __init__(self, firmware_folder: pathlib.Path) -> None:
+        self._firmware_folder: pathlib.Path = firmware_folder
```

Comment:
is this not used?

Reply:
No. It's from a previous implementation. This is handled by FirmwareManager now. Will remove.